src/Symfony/Component/Form/Extension/Core/DataTransformer/DateIntervalToArrayTransformer.php

```diff
@@ -0,0 +1,168 @@
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\DataTransformer;
+
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+
+/**
+ * Transforms between a normalized date interval and a interval string/array.
+ *
+ * @author Steffen Roßkamp <steffen.rosskamp@gimmickmedia.de>
+ */
+class DateIntervalToArrayTransformer implements DataTransformerInterface
+{
+    private $fields;
+    private $availableFields = array(
+        'years' => 'y',
+        'months' => 'm',
+        'days' => 'd',
+        'hours' => 'h',
+        'minutes' => 'i',
+        'seconds' => 's',
+        'invert' => 'r',
+    );
+
+    /**
+     * Constructor.
+     *
+     * @param array $fields The date fields
+     * @param bool  $pad    Whether to use padding
+     */
+    public function __construct(array $fields = null, $pad = false)
+    {
+        if (null === $fields) {
+            $fields = array('years', 'months', 'days', 'hours', 'minutes', 'seconds', 'invert');
+        }
+        $this->fields = $fields;
+        $this->pad = (bool) $pad;
+    }
+
+    /**
+     * Transforms a normalized date interval into a interval array.
+     *
+     * @param  \DateInterval                 $dateInterval Normalized date interval.
+     * @return array                         Interval array.
+     * @throws TransformationFailedException If the given value is not an
+     *                                                    instance of \DateInterval.
+     */
+    public function transform($dateInterval)
+    {
+        if (null === $dateInterval) {
+            return array_intersect_key(
+                array(
+                    'years' => '',
+                    'months' => '',
+                    'weeks' => '',
+                    'days' => '',
+                    'hours' => '',
+                    'minutes' => '',
+                    'seconds' => '',
+                    'invert' => false,
+                ),
+                array_flip($this->fields)
+            );
+        }
+        if (!$dateInterval instanceof \DateInterval) {
+            throw new TransformationFailedException('Expected a \DateInterval.');
+        }
+        $result = array();
+        foreach ($this->availableFields as $field => $char) {
+            $result[$field] = $dateInterval->format('%'.($this->pad ? strtoupper($char) : $char));
+        }
+        if (in_array('weeks', $this->fields, true)) {
+            $result['weeks'] = 0;
+            if (isset($result['days']) && (int) $result['days'] >= 7) {
+                $result['weeks'] = (string) floor($result['days'] / 7);
+                $result['days'] = (string) ($result['days'] % 7);
+            }
+        }
+        $result['invert'] = $result['invert'] === '-' ? true : false;
```

Comment:
just ``$result['invert'] = $result['invert'] === '-'`` will do it. ``$result['invert'] === '-'`` already gives you a boolean